source_code/src/OLEDMINI/bitstreammini.c

```diff
@@ -45,38 +45,36 @@ void miniBistreamInit(bitstream_mini_t* bs, uint8_t height, uint16_t width, uint
     bs->width = width;
     bs->height = height;
     bs->dataCounter = 0;
-    bs->bufferInd = sizeof(bs->buffer);
-    bs->dataSize = (uint16_t)width * (((uint16_t)height+7) >> BITSTREAM_PIXELS_PER_BYTE_BITSHIFT);
+    bs->dataSize = (uint16_t)width * (((uint16_t)height+7) / BITSTREAM_PIXELS_PER_BYTE);
 }
 
 /*! \fn     bsGetNextByte(bitstream_mini_t* bs)
  *  \brief  Return the next data byte from flash
  *  \param  bs      pointer to initialized bitstream context to get the next word from
  *  \return next data word, or 0 if end of data reached
  */
-uint8_t miniBistreamGetNextByte(bitstream_mini_t* bs)
-{
-    // Are you requesting bytes when you've already read everything?
-    if (bs->dataCounter < bs->dataSize) 
-    {
-        bs->dataCounter++;
-        
-        // Check if we need to fetch new data from the SPI flash
-        if(bs->bufferInd >= sizeof(bs->buffer))
-        {
-            // Fetch new data from external flash
-            flashRawRead(bs->buffer, bs->addr, sizeof(bs->buffer));
-            bs->addr += sizeof(bs->buffer);
-            bs->bufferInd = 0;
-            //usbPrintf_P(PSTR("bistream buffer: %02x %02x %02x %02x %02x %02x"), bs->buffer[0], bs->buffer[1], bs->buffer[2], bs->buffer[3], bs->buffer[4], bs->buffer[5]);
-        }
-        
-         return bs->buffer[bs->bufferInd++];
-    }
-    else
-    {
-        return 0;
-    }
-}
+ uint8_t miniBistreamGetNextByte(bitstream_mini_t* bs)
+ {
+     // Are you requesting bytes when you've already read everything?
+     if (bs->dataCounter < bs->dataSize)
+     {
+         uint16_t buffer_index = bs->dataCounter % sizeof(bs->buffer);
```

Comment:
Why? This is simple byte arithmtic? There is not a single reason why it would now take longer. And even if it does, its not a problem unless you can't see it. And the main menu is the only place to see the difference.

What you can try instead is to use a buffer of 1 and then see how a slowdown looks like. Then you can see that there is no slowdown at all.

As I said the % results in & 0x0F. You can simply check that by doing a sha256sum of the hexfile. It is absolutely equal and optimized by the compiler.

I've also noticed that the current spi library has some "faster" functions to read the data faster. Those are also not required and only give additional overhead. I've tested that also and have no problems with the normal functions. I've been using the normal spi functions for 2 weeks now without problems at all. Is there any reason why the display needs to be so fast except the main menu where it is fast enough?